src/js/plugins/plugin.search.js

```diff
@@ -0,0 +1,556 @@
+    queryStringWithBTruncated = queryString
+      .replace(/^(.{100}[^\s]*).*/, "$1")
+      .replace(re, '<b>$1</b>')
+            + '...';
+  } else {
+    queryStringWithBTruncated = queryString.replace(re, '<b>$1</b>');
+  }
+
+  // draw marker
+  const markerTopValue = (-this.refs.$brContainer.height());
+  $('<div>')
+    .addClass('BRsearch')
+    .css({
+      top: `${markerTopValue}px`,
+      left: percentThrough,
+    })
+    .attr('title', uiStringSearch)
+    .append(
+      $('<div>')
+        .addClass('BRquery')
+        .append(
+          $('<div>').html(queryStringWithB),
+          $('<div>').html(`${uiStringPage} ${pageNumber}`)
+        )
+    )
+    .data({ pageIndex })
+    .appendTo(this.$('.BRnavline'))
+    .bt(searchBtSettings)
+    .hover(
+      (event) => {
+        // remove from other markers then turn on just for this
+        // XXX should be done when nav slider moves
+        $('.BRsearch,.BRchapter').removeClass('front');
+        $(event.target).addClass('front');
+      },
+      (event) => $(event.target).removeClass('front'))
+    .click(function (event) {
+      // closures are nested and deep, using an arrow function breaks references.
+      // Todo: update to arrow function & clean up closures
+      // to remove `bind` dependency
+      this.jumpToIndex($(event.target).data('pageIndex'));
+    }.bind(this))
+    .animate({top:'-25px'}, 'slow');
+
+  // Add Mobile Search Results
+  const imgPreviewUrl = this.getPageURI(pageIndex, 16, 0); // scale 16 is small
+  const $mobileSearchResultWrapper = this.$('.BRmobileSearchResultWrapper');
+  if ($mobileSearchResultWrapper.length) {
+    const onResultsClick = (e) => {
+      e.preventDefault();
+      this.switchMode(this.constMode1up);
+      this.jumpToIndex(pageIndex);
+      this.refs.$mmenu.data('mmenu').close();
+    };
+    $(
+      `<a class="BRmobileSearchResult">
+        <table>
+          <tr>
+            <span class="pageDisplay">${pageDisplayString}</span>
+          </tr>
+          <tr>
+            <td>
+              <img class="searchImgPreview" src="${imgPreviewUrl}" />
+            </td>
+            <td>
+              <span>${queryStringWithBTruncated}</span>
+            </td>
+          </tr>
+        </table>
+      </a>`
+    )
+      .attr('href', '#search/' + this.searchTerm)
+      .click(onResultsClick)
+      .appendTo($mobileSearchResultWrapper)
+    ;
+  }
+};
+
+/**
+ * Removes all search pins
+ */
+BookReader.prototype.removeSearchResults = function() {
+  this.removeSearchHilites(); //be sure to set all box.divs to null
+  this.searchTerm = null;
+  this.searchResults = null;
+  this.trigger(BookReader.eventNames.fragmentChange);
+  this.$('.BRnavpos .BRsearch').remove();
+  this.$('.BRmobileSearchResultWrapper').empty(); // Empty mobile results
+};
+
+/**
+ * Returns true if a search highlight is currently being displayed
+ */
+BookReader.prototype.searchHighlightVisible = function() {
```

Comment:
lets add jsdocs include `@return { ... }`

Reply:
thanks for catching. added